### PR TITLE
[python] update numpy version

### PR DESCRIPTION
**Pull request by timothytrippel**
The numpy version that was added in #15178 is too old, and is not compatable with newer version of Python, e.g, 3.10. Since we upgraded CI to use Ubunut 20.04, which ships with Python 3.8.5, numpy version 1.22.0 should be compatable with both the officially supported Python version, and more modern versions of Python, e.g., 3.10.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

**Comment by timothytrippel** [comment]
looks like we may need to upgrade private CI VMs to 20.04 as well, if this is possible? @GregAC @mundaym @jprendes

**Review by alphan — APPROVED** [review]

**Comment by alphan** [comment]
(assuming we can resolve the CI issues)

**Comment by gdessouky** [comment]
I tried with numpy version 1.22.0 and confirm that it works locally with Python 3.10.7 on my Google provisioned machine.

**Review by gdessouky — APPROVED** [review]
I tried with numpy version 1.22.0 per this commit and confirm that it works locally with Python 3.10.7 on my Google provisioned machine.

**Comment by alphan** [comment]
> Confirmed with @jprendes, he is going to be upgrading the remaining CI machines to Ubuntu 20.04, and then this should pass CI.

Thanks @timothytrippel and @jprendes !

**Comment by vogelpi** [comment]
@vrozic and I discussed to just again drop the numpy dependency to unblock this. Because at the moment, the topgen integration of strong random (which requires #15178) has been postponed. Vladimir, can you please comment on this and do a PR as discussed?

**Comment by vrozic** [comment]
Indeed, I've added numpy to the requirements to enable strong_random integration.
In the meantime we've decided to move this integration to M3, so numpy is not needed at the moment.
I will quickly make the PR to remove it from the `python-requirements.txt`

**Comment by vogelpi** [comment]
That would be great, thanks @vrozic !

**Comment by vrozic** [comment]
I've temporarily removed numpy requirement in #15875 
It will be needed again once we start working on integrating strong_random class, but that can wait until all CI machines are upgraded.

**Comment by timothytrippel** [comment]
SG, thanks @vrozic and @vogelpi ! I will close this PR then.

**Pull request closed without merging**
